### ISSUE TITLE
gnmic retries stream subscriptions after INVALID_ARGUMENT error

### ISSUE THREAD

**Issue by cmcdona2** [openconfig/gnmic#516]
Hi gnmic devs,

I'm using an OpenConfig server which (reasonably) returns a gRPC INVALID_ARGUMENT / error code 3 error when I request `/system/|_@~\\` from it (regardless of whether I'm doing a `get` or a `subscribe`, and regardless of the `--stream-mode`). Using `gnmic` to make the request, if I use `get` then I see the INAVLID_ARGUMENT error returned. However, if I use `subscribe` then I don't see any error. Internally (from within the server) I can see it logs the invalid request every 10s (probably because of the default retry period as per `--retry duration ... retry timer for RPCs (default 10s)`) and the INVALID_ARGUMENT error that it's returning. I think `gnmic` should return instead of retrying if it sees that a subscription attempt returned INVALID_ARGUMENT (because the arguments aren't going to become valid on retries).

Thank you,
Colin

**Comment by karimra** [comment]
Yes, gNMIc does not look into the error code for Subscribe. That's something that can be enhanced.
